app/services/atmosphere/cloud/vm_creator.rb

```diff
@@ -27,7 +20,26 @@ def execute
         server_params[:nics] = [{ net_id: @nic }]
       end
 
-      set_security_groups!(server_params)
+      case (@tmpl.compute_site.technology)
+      when 'azure'
+        server_params[:vm_name] = SecureRandom.uuid
+        server_params[:vm_user] = Atmosphere.azure_vm_user
+        server_params[:password] = Atmosphere.azure_vm_password
```

Comment:
Hound has had a similar comment. Suggest a better solution. I'm the author of this part so I'm very subjective about understanding this code.